### PR TITLE
Make project header version-agnostic

### PR DESCRIPTION
**Pull request by aeolianeth**
- break into multiple components
- pass props instead of using v1 context

The motivation is to enable us to use this component for v2 projects.

## Acceptance checklist

- [x] I have evaluated the [Approval Guidelines](../CONTRIBUTING.md#approval-guidelines) for this PR.
- [x] I have tested this PR in [all supported browsers](../CONTRIBUTING.md#browser-support).

### CODE REVIEW

File: src/components/v1/V1Project/V1ProjectHeaderActions.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
+import { t, Trans } from '@lingui/macro'
```
Comment (aeolianeth) [inline src/components/v1/V1Project/V1ProjectHeaderActions.tsx:1]: **note**: you can imagine us having a v2 version of this component.

**Comment by aeolianeth** [comment]
@johnnyd-eth @peripheralist this is another set of cherry-picks from https://github.com/jbx-protocol/juice-interface/pull/551.

Let me know what you think 😊

**Review by johnnyd-eth — APPROVED** [review]
Niceee cleanup here @aeolianeth. Nothing from me. Approving

**Comment by aeolianeth** [comment]
Thanks @johnnyd-eth. I've tested the project actions (updating metadata, adding to balance) locally and everything appears unaffected. The project header itself looks good on Fleek.

Merging 🚀

**Pull request merged**